demo/util/demo-controller.mjs

```diff
@@ -0,0 +1,124 @@
+import * as Util from './util.mjs';
+
+/**
+ * Create a string of |wordCount} words.
+ */
+function words(wordCount) {
+  let result = '';
+  for (let i = 0; i < wordCount; i++) {
+    result += ' word';
+  }
+  return result;
+}
+
+const NUM_WORDS = 50;
+const WORDS = words(NUM_WORDS);
+const DEFAULT_COUNT = 10000;
+
+
+const TEMPLATE = `
+<div id=status>
+</div>
+<div id=buttons>
+  <div>
+    items:
+    <button class=count>1000</button>
+    <button class=count>10000</button>
+    <button class=count>100000</button>
+  </div>
+  <button id=swap></button>
+</div>
+`;
+
+/**
+ * Displays buttons that change the number of elements in a container
+ * and also whether the container is a vitual-scroller or a plain
+ * div. It also displays the current count and container type.
+ */
+export class DemoController extends HTMLElement {
+  #count = DEFAULT_COUNT;
+  #container;
+
+  #swapButton;
+  #statusDiv;
+
+  constructor() {
+    super();
+
+    const shadowRoot = this.attachShadow({mode: 'closed'});
+    shadowRoot.innerHTML = TEMPLATE;
+
+    const buttons = shadowRoot.getElementById('buttons');
+    for (const button of buttons.getElementsByClassName('count')) {
+      button.onclick = e => {
+        this.#setCount(parseInt(e.target.textContent));
+      };
+    }
+
+    this.#swapButton = shadowRoot.getElementById('swap');
+    this.#swapButton.onclick = () => {
+      this.#swap();
+    };
+    this.#statusDiv = shadowRoot.getElementById('status');
+  }
+
+  /**
+   * Set the container that will be managed by the controller.
+   */
+  setContainer(container) {
+    this.#container = container;
+    this.#updateContainer();
+  }
+
+  #setCount = count => {
+    this.#count = count;
+    this.#updateContainer();
+  }
+
+  #swapTo = () => {
```

Comment:
Naming implies it'll actually swap; maybe getSwappedLocalName or something

Reply:
done.